umbral/openssl.py

```diff
@@ -20,67 +22,62 @@ def _get_new_BN(set_consttime_flag=True):
 
 def _get_ec_group_by_curve_nid(curve_nid: int):
     """
-    Returns the group of a given curve via its OpenSSL nid.
+    Returns the group of a given curve via its OpenSSL nid. This must be freed
+    after each use otherwise it leaks memory.
     """
     group = backend._lib.EC_GROUP_new_by_curve_name(curve_nid)
     backend.openssl_assert(group != backend._ffi.NULL)
 
```

Comment:
Isn't here missing an `EC_GROUP_clear_free`?

Reply:
If you call it, it gets cleared from memory immediately. Apparently, the garbage collector picks it up right then.

I think I would add it to a `__del__` method, but not really sure what the benefits are.